File: carddemo/app/models/user_security.py
"""User security model — from CSUSR01Y.cpy (80-byte VSAM KSDS, key=SEC-USR-ID 8 chars).

COBOL record layout:
  05 SEC-USR-ID      PIC X(08)   → String(8) PK
  05 SEC-USR-FNAME   PIC X(20)   → String(20)
  05 SEC-USR-LNAME   PIC X(20)   → String(20)
  05 SEC-USR-PWD     PIC X(08)   → stored as bcrypt hash (String(128))
  05 SEC-USR-TYPE    PIC X(01)   → String(1)  'A'=admin, 'U'=regular
  05 SEC-USR-FILLER  PIC X(23)   → not stored

COBOL 88-level conditions:
  88 CDEMO-USRTYP-ADMIN VALUE 'A'  → UserType.ADMIN
  88 CDEMO-USRTYP-USER  VALUE 'U'  → UserType.USER
"""

import enum

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class UserType(str, enum.Enum):
    ADMIN = "A"
    USER = "U"


class UserSecurity(Base):
    __tablename__ = "user_security"

    usr_id: Mapped[str] = mapped_column(String(8), primary_key=True)
    usr_fname: Mapped[str] = mapped_column(String(20), default="")
    usr_lname: Mapped[str] = mapped_column(String(20), default="")
    usr_pwd: Mapped[str] = mapped_column(String(128), default="")
    usr_type: Mapped[str] = mapped_column(String(1), default="U")

File: java/carddemo-common/src/main/java/com/carddemo/common/model/UserSecurity.java
package com.carddemo.common.model;

import com.carddemo.common.model.enums.UserType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * JPA entity mapping for the User Security VSAM record.
 * Maps to COBOL copybook: CSUSR01Y.cpy (80-byte record)
 *
 * COBOL record layout:
 *   SEC-USR-ID      PIC X(08)
 *   SEC-USR-FNAME   PIC X(20)
 *   SEC-USR-LNAME   PIC X(20)
 *   SEC-USR-PWD     PIC X(08)
 *   SEC-USR-TYPE    PIC X(01)
 *
 * The userType maps to the 88-level conditions in COCOM01Y.cpy:
 *   CDEMO-USRTYP-ADMIN VALUE 'A'
 *   CDEMO-USRTYP-USER  VALUE 'U'
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "user_security")
public class UserSecurity {

    @Id
    @Column(name = "user_id", length = 8, nullable = false)
    private String userId;

    @Column(name = "first_name", length = 20)
    private String firstName;

    @Column(name = "last_name", length = 20)
    private String lastName;

    @Column(name = "password", length = 8)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(name = "user_type", length = 5)
    private UserType userType;
}
